warehouse/query-core/src/main/java/datawave/query/jexl/JexlNodeBuilder.java

```diff
@@ -0,0 +1,637 @@
+import org.apache.commons.jexl2.parser.ASTFalseNode;
+import org.apache.commons.jexl2.parser.ASTForeachStatement;
+import org.apache.commons.jexl2.parser.ASTFunctionNode;
+import org.apache.commons.jexl2.parser.ASTGENode;
+import org.apache.commons.jexl2.parser.ASTGTNode;
+import org.apache.commons.jexl2.parser.ASTIdentifier;
+import org.apache.commons.jexl2.parser.ASTIfStatement;
+import org.apache.commons.jexl2.parser.ASTJexlScript;
+import org.apache.commons.jexl2.parser.ASTLENode;
+import org.apache.commons.jexl2.parser.ASTLTNode;
+import org.apache.commons.jexl2.parser.ASTMapEntry;
+import org.apache.commons.jexl2.parser.ASTMethodNode;
+import org.apache.commons.jexl2.parser.ASTModNode;
+import org.apache.commons.jexl2.parser.ASTMulNode;
+import org.apache.commons.jexl2.parser.ASTNENode;
+import org.apache.commons.jexl2.parser.ASTNRNode;
+import org.apache.commons.jexl2.parser.ASTNotNode;
+import org.apache.commons.jexl2.parser.ASTNullLiteral;
+import org.apache.commons.jexl2.parser.ASTNumberLiteral;
+import org.apache.commons.jexl2.parser.ASTOrNode;
+import org.apache.commons.jexl2.parser.ASTReference;
+import org.apache.commons.jexl2.parser.ASTReferenceExpression;
+import org.apache.commons.jexl2.parser.ASTReturnStatement;
+import org.apache.commons.jexl2.parser.ASTSizeFunction;
+import org.apache.commons.jexl2.parser.ASTSizeMethod;
+import org.apache.commons.jexl2.parser.ASTStringLiteral;
+import org.apache.commons.jexl2.parser.ASTTernaryNode;
+import org.apache.commons.jexl2.parser.ASTTrueNode;
+import org.apache.commons.jexl2.parser.ASTUnaryMinusNode;
+import org.apache.commons.jexl2.parser.ASTVar;
+import org.apache.commons.jexl2.parser.ASTWhileStatement;
+import org.apache.commons.jexl2.parser.JexlNode;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Supplier;
+
+/**
+ * A builder class that supports fluent-style building of {@link JexlNode} instances.
+ * 
+ * @param <T>
+ *            the {@link JexlNode} type
+ */
+public class JexlNodeBuilder<T extends JexlNode> {
+    
+    private final Supplier<T> constructor;
+    private final List<Supplier<? extends JexlNode>> children = new ArrayList<>();
+    private String image;
+    private Object value;
+    
+    public JexlNodeBuilder(JexlNodeInstance<T> instance) {
+        this.constructor = instance.getConstructor();
+    }
+    
+    /**
+     * Set the image for the node this builder will create.
+     * 
+     * @param image
+     *            the image
+     * @return this builder
+     */
+    public JexlNodeBuilder<T> withImage(String image) {
+        this.image = image;
+        return this;
+    }
+    
+    /**
+     * Set the value for the node this builder will create.
+     * 
+     * @param value
+     *            the value
+     * @return this builder
+     */
+    public JexlNodeBuilder<T> withValue(Object value) {
+        this.value = value;
+        return this;
+    }
```

Comment:
Should we consider throwing an exception on these methods if they are called when creating a JexlNode type which doesn't have an image or value?

Reply:
For example, and node, or node, etc.

Reply:
An interesting problem, and I'm inclined to say yes. I will also experiment with different design approaches to see if we can make it so that the methods `withImage()` and `withValue` are non-existent when creating nodes for which they're not supported.

Can you provide a list of what node types can/cannot have images/values, whichever one is shorter?

Reply:
I don't know if I can make a comprehensive list, but the first few that come to mind that have neither an image or value are the AND/OR nodes.

Reply:
Okay, understood.